src/Storages/StorageDictionary.cpp

```diff
@@ -211,6 +211,8 @@ void StorageDictionary::renameInMemory(const StorageID & new_table_id)
 {
     auto old_table_id = getStorageID();
     IStorage::renameInMemory(new_table_id);
+    bool move_between_db_engines = old_table_id.uuid == UUIDHelpers::Nil || new_table_id.uuid == UUIDHelpers::Nil;
```

Comment:
But what if both are `Nil`? `assert` below says that it's impossible, but maybe add a comment?

Reply:
Maybe we can allow renaming dictionaries in `Ordinary` database, I will check.